### DISCUSSION TITLE
Passing reactive object as additional argument to several golem modules

### DISCUSSION THREAD

**Discussion by agronomofiorentini** [ThinkR-open/golem#739] [Q&A] [answered]
Hi ThinkR,
Thank you for the development of this amazing package which I always use when I am faced with the development of a very large web application.
I would like to pass to several golem module an additional argument which is the reactive object of check_credentials of shiny manager to:
1.	To call only the user data.
2.	To activate or deactivate based on the type of user

This is the way that i am operating right now, but finding many problems.

UI part

`app_ui<-shinymanager::secure_app(app_ui, status="primary")`

Server Part

`res_auth <- secure_server(check_credentials = check_credentials(credentials))
  auth_output<- reactive({
    reactiveValuesToList(res_auth)})`

Module Server
`mod_golem_User("golem_ui_1", res_auth)`

Then when i am calling the res_auth() in the golem module it is as if it did not exist

**Comment by ColinFay** [comment]
Hey, 

These two issues seems to be unrelated :) 
Please open one issue by topic so that we can keep track of everything 🤘

Can you please move the Docker related issue to [`{dockerfiler}`](https://github.com/ColinFay/dockerfiler/issues/new) please ? This is where all the docker related elements are now stored :) 

The original post being a question, I'll convert this issue to a Discussion :) 

Thanks for using `{golem}`! 

Cheers, 
Colin

**Reply by agronomofiorentini** [reply]
Hi @ColinFay it is a pleasure to talk with you and also with @VincentGuyader. I see many videos of you explain how to use golem.

I have reported the dockerfile issue as you suggested to me.

I would ask to you if there are any method to pass a reactivevalue from the server.R (which in this case is the shinymanager reactivevalue output) to any golem_module in order to keep the modularization safe. 

if there is no possibility to do this, the modularization of the code according to the user's information becomes difficult.

**Comment by ColinFay** [comment, marked as answer]
Hey, 

There is an example here about `{shinymanager}` and `{golem}` that does reuse `res_auth` https://github.com/ColinFay/golemexamples/blob/master/golemshinymanager/R/app_server.R#L27

Let me know if that solves your issue.

Cheers, 
Colin

**Reply by agronomofiorentini** [reply]
It solved my problem.

I knew you could give me a solution
thank you.

I wish you a good day.

Marco
